ckanext/geodatagov/cli.py

```diff
@@ -344,24 +369,50 @@ def db_solr_sync(dryrun, cleanup_solr, update_solr):
 
     # get active packages from DB
     active_package = [
-        (r[0], r[1].replace(microsecond=0))
-        for r in model.Session.query(model.Package.id, model.Package.metadata_modified)
-        .filter(model.Package.state != "deleted")
+        (r[0], r[1].replace(microsecond=0), r[2])
+        for r in model.Session.query(
+            model.Package.id,
+            model.Package.metadata_modified,
+            HarvestObject.id
+        )
+        .join(
+            HarvestObject,
+            and_(
+                HarvestObject.package_id == model.Package.id,
+                HarvestObject.current == True  # noqa: E712
```

Comment:
Does `HarvestObject.current` not work as a boolean value by itself?  Or are you trying to make it more readable?

Reply:
got it from upsteam; and it is more readable.